src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3561,9 +3560,12 @@ void ride_construction_toolupdate_construct(ScreenCoordsXY screenCoords)
     }
 
     _previousTrackPiece = _currentTrackBegin;
+    // search for appropriate z value for ghost, up to max ride height
+    int numAttempts = (z <= (8 * MAX_TRACK_HEIGHT) ? MAX_TRACK_HEIGHT - (z / 8) + 1 : 2);
+
     if (ride->type == RIDE_TYPE_MAZE)
     {
-        for (;;)
+        for (int zAttempts = 1; zAttempts < numAttempts; ++zAttempts)
```

Comment:
Whats the reason here to start at 1? The other loops seem to be starting at 0

Reply:
The previous code had:
```
bx--;
if (bx == 0)
    break;
```
So it meant that the `currentTrackBegin.z += 16` was only done `bx - 1` times. I guess that's why the for starts at `1`

Reply:
Something doesn't quite add up, a few lines below it also had the check for when ```bx >= 0``` which is impossible to reach given that it would break out if ```bx == 0```. Can you test what happens if you were to start every loop at 0?

Reply:
Seems to not make a difference, at least in what I'm doing. I'm using `Raceway Park` and testing building above the tree line.